src/main/java/application/models/User.java

```diff
@@ -93,4 +93,28 @@ public void addFollower(User follower){
     public void removeFollower(User follower){
         this.followersList.remove(follower);
     }
+
+    public String getDegreeOfSeparation(User user){
+         Set<User> userList1st = new HashSet<>();
+        userList1st.addAll(this.followersList);
+        userList1st.addAll(this.followingList);
+
+         if(userList1st.contains(user)){
+             return "1st";
+         }
+         else{
+             Set<User> userList2nd = new HashSet<>();
+             for (User userInList:
+                 userList1st ) {
+                 userList2nd.add((User) userInList.getFollowersList());
+                 userList2nd.add((User) userInList.getFollowingList());
+             }
+
+             if(userList2nd.contains(user)){
+                 return "2nd";
+             }
+         }
+
+        return "3rd+";
+    }
```

Comment:
We should probably go past just two degrees of separation (most likely up to 6 considering "6 degrees of kevin bacon").  I would also recommend writing this a little more "expandably". Could probably also use recursion, but there's not much need.  For example (assume MAX_DEGREES is declared already):

```
public int getDegreesOfSeparation(User user) {
  Set<User> directAcquaintances = new HashSet<>();
  Set<User> checked = new HashSet<>();
  directAcquaintances.addAll(this.followersList);
  directAcquaintances.addAll(this.followingList);
  for(int degrees = 1; degrees <= MAX_DEGREES; degrees++) {
    if(directAcquaintances.contains(user)) {
      return degrees;
    }
    else {
      checked.addAll(directAcquaintances);
      for(User u : new HashSet<>(directAcquaintances)) {
        directAcquaintances.addAll(u.getFollowersList);
        directAcquaintances.addAll(u.getFollowingList);
      }
      directAcquaintances.removeAll(checked);
  }
  return -1 // Signals that its more than MAX_DEGREES of separation
}
```

Probably some errors with this but for a general idea.

Reply:
sounds good, I've updated the method!